include/mbedtls/ssl.h

```diff
@@ -3602,10 +3602,10 @@ int mbedtls_ssl_get_record_expansion( const mbedtls_ssl_context *ssl );
 
 /**
  * \brief          Return the current maximum outgoing record payload in bytes.
- *                 This takes into account the config.h setting \c
- *                 MBEDTLS_SSL_OUT_CONTENT_LEN, the configured and negotiated
- *                 max fragment length extension if used, and for DTLS the
- *                 path MTU as configured and current record expansion.
+ *                 This takes into account various factors, such as the config.h
+ *                 setting \c MBEDTLS_SSL_OUT_CONTENT_LEN, extensions such as the
+ *                 max fragment length or record size limit extension if used, and
+ *                 for DTLS the path MTU as configured and current record expansion.
```

Comment:
I have similar reservations as in the PSK PR: I don't think we want to start being vague in the documentation, but instead clearly describe the current behaviour and how it might evolve future. Though here to be clear, we might want to put that in a note rather than the main description, which should indeed remain general/abstract, for example:

```
/**
 * \brief Return the current maximum outgoing record payload in bytes.
 *
 * \note Currently this depends on x, y and z. In future versions of the library,
 * it may depend also on other factors such as t and u.
```

Reply:
I don't think we're being vague here. The headline clearly says what this function returns, and the text gives some more context. I am happy to put this text in a `\note`, though.

Reply:
I'm still not a huge fan of the new wording, I think "version-specific" gives an impression of uncertainty to the reader, and I'd prefer the order "This is how it works now. This is what may change in the future." (as is, present before future, certain before uncertain). But since we already have a clear definition of what the function does, and this is only a note about how it does it, I don't think it's worth agonizing over it, the present version if good enough and we can always tweak it later if we really want to.